### PR TITLE
wgsl: Fix field semantics even if input struct doesn't need flattening

### PR DESCRIPTION
**Pull request by aleino-nv**
This fixes an issue where location indices of stage inputs were repeated

https://github.com/shader-slang/slang/issues/5633

### CODE REVIEW

**Review by csyonghe — APPROVED** [review]

**Review by jkwak-work — CHANGES_REQUESTED** [review]
I feel uncomfortable with this change.

The code is copied exactly from slang-ir-metal-legalize.cpp
And with this PR, WGPU side works but Metal side is failing.

I suggest to investigate why Metal is failing and fix it properly before merging this change.

**Comment by csyonghe** [comment]
I feel like metal needs the same treatment.

File: source/slang/slang-ir-wgsl-legalize.cpp [hunk 1]
```diff
@@ -145,11 +145,10 @@ struct LegalizeWGSLEntryPointContext
                     structType,
                     mapOldFieldToNewField,
                     semanticInfoToRemove);
+                // Validate/rearange all semantics which overlap in our flat struct.
+                fixFieldSemanticsOfFlatStruct(flattenedStruct);
```
Comment (jkwak-work) [inline source/slang/slang-ir-wgsl-legalize.cpp:149]: If this function is going to be called even when it is not flattened, the name of the function may need to change to match

**Comment by jkwak-work** [comment]
> I feel like metal needs the same treatment.

Yes, that must be it.

**Review by jkwak-work — APPROVED** [review]

**Comment by aleino-nv** [comment]
> > I feel like metal needs the same treatment.
> 
> Yes, that must be it.

**Comment by aleino-nv** [comment]
> > I feel like metal needs the same treatment.
> 
> Here is a PR #5666

Ok the PR fails, I filed https://github.com/shader-slang/slang/issues/5667.
The render0 test is also broken for Metal, so it's probably best to fix that one first, and then enable this test.

**Pull request merged**